src/deluge/processing/engines/audio_engine.cpp

```diff
@@ -953,6 +953,7 @@ void routine() {
 
 	if (audioRoutineLocked) {
```

Comment:
could this check become a run condition using the logic added for song loading instead?

Reply:
Not yet, when the CPU is busy run conditions aren't checked often enough for the audio routine since checking them is the lowest possible priority. 

Something similar should be doable but not possible yet